src/lib/voiceSearchHelper/types.ts

```diff
@@ -0,0 +1,34 @@
+export type Status =
+  | 'initial'
+  | 'askingPermission'
+  | 'waiting'
+  | 'recognizing'
+  | 'finished'
+  | 'error';
+
+export type VoiceListeningState = {
+  status: Status;
+  transcript: string;
+  isSpeechFinal: boolean;
+  errorCode?: string;
```

Comment:
When we expose a value that is explicitly not defined in our codebase, we usually set it to `null` ([examples](https://github.com/algolia/instantsearch.js/blob/1a6dc518763132eb3223c078cb38841bca83b602/src/widgets/index/index.ts#L47-L65)).

I think we should do the same here.

Reply:
I don't know what you're talking about, error code is defined in our implementation, but since we don't use it ourselves (only exposed to users, so breaking change to change that), it's either a string or undefined. This is not a scoped variable, but just some key in an object.

Do I misunderstand what you're suggesting?

Reply:
I meant that it would have made sense to have it be a `string | null` but didn't realise it's already exposed to user, so breaking change 👍